libglusterfs/src/mem-pool.c

```diff
@@ -463,29 +455,37 @@ pool_sweeper(void *arg)
     pending = true;
 
     for (;;) {
+        int ret;
+        struct timespec ts;
+
+        timespec_now_realtime(&ts);
```

Comment:
nit: timespec_now() seems good enough here?

Reply:
No. `pthread_cond_timedwait()` expects absolute time, but `timespec_now()` uses CLOCK_MONOTONIC which is relative to, as described by POSIX, "some unspecified point in the past" (on Linux, an actual "point" is system boot time).

Reply:
Apologies, I think gf_time() then.